src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs

```diff
@@ -403,6 +403,8 @@ fromForeignPtrUnsafe ::
     Storable a =>
 #endif
     ForeignPtr a -> Ptr a -> Ptr a -> Array a
+fromForeignPtrUnsafe (ForeignPtr start _) _ _
+    | Ptr start == nullPtr = Array nilArrayContents nullPtr nullPtr nullPtr
```

Comment:
We can add a `nil` array API as we have in `src/Streamly/Internal/Data/Array/Foreign/Type.hs`.